be/src/vec/olap/vcollect_iterator.cpp

```diff
@@ -260,6 +260,14 @@ VCollectIterator::Level1Iterator::~Level1Iterator() {
             child = nullptr;
         }
     }
+
+    if (_heap) {
+        while (!_heap->empty()) {
+            auto child = _heap->top();
+            _heap->pop();
+            if (child) delete child;
```

Comment:
all elements in _heap are also in _children. Already delete in line 257, maybe double delete here.

Reply:
at line 327, _children.clear() is called after child is added to _heap.